### PR TITLE
[jest] update jest.clearAllMocks() description and remove @deprecated

### PR DESCRIPTION
**Pull request by a-xin**
Update the description of `jest.clearAllMocks()` to the one from:
https://facebook.github.io/jest/docs/en/jest-object.html#jestclearallmocks

This also **removes** the incorrect `@deprecated` annotation: The docs don't mention that `clearAllMocks()` is deprecated.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- ~~Increase the version number in the header if appropriate.~~
- ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~

### CODE REVIEW

**Comment by a-xin** [comment]
Looks like the bot currently doesn't trigger notifications, therefore pinging manually:

:bell: @NoHomey @jwbay @asvetliakov @alexjoverm @epicallan @ikatyang @wsmd @JamieMason @douglasduteil - please review this PR in the next few days. Be sure to explicitly select Approve or Request Changes in the GitHub UI so I know what's going on.

If no reviewer appears after a week, a DefinitelyTyped maintainer will review the PR instead.

**Review by wsmd — APPROVED** [review]
Thank you!

**Comment by andy-ms** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
